internal/commands/user/disable_test.go

```diff
@@ -0,0 +1,324 @@
+package user
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/app"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+)
+
+func TestDisableSetup(t *testing.T) {
+	t.Run("should construct a realm client with the configured base url", func(t *testing.T) {
+		profile := mock.NewProfile(t)
+		profile.SetRealmBaseURL("http://localhost:8080")
+		cmd := &CommandDisable{inputs: disableInputs{}}
+
+		assert.Nil(t, cmd.realmClient)
+		assert.Nil(t, cmd.Setup(profile, nil))
+		assert.NotNil(t, cmd.realmClient)
+	})
+}
+
+func TestDisableHandler(t *testing.T) {
+	projectID := "projectID"
+	appID := "appID"
+	testApp := realm.App{
+		ID:          appID,
+		GroupID:     projectID,
+		ClientAppID: "eggcorn-abcde",
+		Name:        "eggcorn",
+	}
+	testUsers := []realm.User{
+		{
+			ID:         "user-1",
+			Identities: []realm.UserIdentity{{ProviderType: realm.AuthProviderTypeAnonymous}},
+			Disabled:   false,
+		},
+		{
+			ID:         "user-2",
+			Identities: []realm.UserIdentity{{ProviderType: realm.AuthProviderTypeAnonymous}},
+			Disabled:   true,
+		},
+	}
+
+	t.Run("should disable a user when a user id is provided", func(t *testing.T) {
+		var capturedAppFilter realm.AppFilter
+		var capturedFindProjectID, capturedFindAppID string
+		var capturedDisableProjectID, capturedDisableAppID string
+		realmClient := mock.RealmClient{}
+		realmClient.FindAppsFn = func(filter realm.AppFilter) ([]realm.App, error) {
+			capturedAppFilter = filter
+			return []realm.App{testApp}, nil
+		}
+		realmClient.FindUsersFn = func(groupID, appID string, filter realm.UserFilter) ([]realm.User, error) {
+			capturedFindProjectID = groupID
+			capturedFindAppID = appID
+			return testUsers[:1], nil
+		}
+		realmClient.DisableUserFn = func(groupID, appID, userID string) error {
+			capturedDisableProjectID = groupID
+			capturedDisableAppID = appID
+			return nil
+		}
+		cmd := &CommandDisable{
+			inputs: disableInputs{
+				ProjectInputs: app.ProjectInputs{
+					Project: projectID,
+					App:     appID,
+				},
+				Users: []string{testUsers[0].ID},
+			},
+			realmClient: realmClient,
+		}
+
+		assert.Nil(t, cmd.Handler(nil, nil))
+		assert.Equal(t, realm.AppFilter{App: appID, GroupID: projectID}, capturedAppFilter)
+		assert.Equal(t, projectID, capturedFindProjectID)
+		assert.Equal(t, appID, capturedFindAppID)
+		assert.Equal(t, projectID, capturedDisableProjectID)
+		assert.Equal(t, appID, capturedDisableAppID)
+		assert.Equal(t, testUsers[0].ID, cmd.inputs.Users[0])
+		assert.Nil(t, cmd.outputs[0].err)
+		assert.Equal(t, cmd.outputs[0].user, testUsers[0])
```

Comment:
[nit] can consolidate these last two asserts with:

```go
assert.Equal(t, userOutput{testUsers[0], nil}
```

if you don't go that route, at least swap the position here so its:

```go
assert.Equal(t, testUsers[0], cmd.outputs[0].user)
```